fpna/DataModel/DataStructure.ipynb

```diff
@@ -0,0 +1,253 @@
+                "#!import \"../Constants/Ifrs17CalculationEngine\"",
+                "\n#!import \"../Constants/Enums\""
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "using Systemorph.Vertex.Grid.Model;",
+                "\nusing Systemorph.Vertex.Pivot.Builder.Interfaces;",
+                "\nusing Systemorph.Vertex.Hierarchies;"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "markdown",
+            "source": [
+                "# Dimensions"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "markdown",
+            "source": [
+                "## AmountType"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public record RatioAmountType : AmountType {}",
+                "\npublic record PatternAmountType : AmountType {}",
+                "\npublic record AssetAmountType : AmountType {}",
+                "\npublic record SolvencyBsAmountType : AmountType {}",
+                "\npublic record IfrsBsAmountType : AmountType {}"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "markdown",
+            "source": [
+                "## Cashflow Definition"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public record CashflowDefinition : IKeyed",
+                "\n{",
+                "\n    [Key]",
+                "\n    [NotVisible]     ",
+                "\n    public Guid Id { get; init; }",
+                "\n",
+                "\n    [IdentityProperty]",
+                "\n    [Dimension(typeof(string), nameof(DataNode))]",
+                "\n    public string DataNode { get; init; }",
+                "\n",
+                "\n    [IdentityProperty]",
+                "\n    [Dimension(typeof(AmountType))] ",
+                "\n    public string AmountType { get; init; }",
+                "\n",
+                "\n    [NoArithmetics(ArithmeticOperation.Scale)]",
+                "\n    [Dimension(typeof(int), nameof(AccidentYear))]",
+                "\n    [IdentityProperty]",
+                "\n    public int? AccidentYear { get; init; }",
+                "\n",
+                "\n    [IdentityProperty]",
+                "\n    [Dimension(typeof(Shape), nameof(Shape))]",
+                "\n    public Shape Shape { get; init; }",
```

Comment:
Maybe Shape is not the best name... but Ok for now.

Reply:
added a TODO to rename this later